src/lower.c

```diff
@@ -10805,8 +10805,28 @@ lower_unit_t *lower_instance(lower_unit_t *parent, cover_tagging_t *cover,
 
    lower_unit_t *lu = lower_unit_new(parent, vu, cover, block);
 
-   if (cover_enabled(lu->cover, COVER_MASK_ALL))
+   if (cover_enabled(lu->cover, COVER_MASK_ALL)) {
+
+      // Need to find corresponding T_HIER in advance before hier
+      // of lower_scope_t is assigned.
+      tree_t inst_hier = NULL;
+      for (int i = 0; i < tree_decls(block); i++) {
+         tree_t decl = tree_decl(block, i);
+         if (tree_kind(decl) == T_HIER) {
+            inst_hier = decl;
+            break;
+         }
+      }
+      assert(inst_hier != NULL);
```

Comment:
It's always the first declaration so you can just do:

```suggestion
      tree_t hier = tree_decl(block, 0);
      assert(tree_kind(hier) == T_HIER);
```